app/celery/letters_pdf_tasks.py

```diff
@@ -524,3 +525,37 @@ def replay_letters_in_error(filename=None):
                     [filename],
                     queue=QueueNames.LETTERS
                 )
+
+
+@notify_celery.task(name='resanitise-pdf')
+def resanitise_pdf(notification_id):
+    """
+    `notification_id` is the notification id for a PDF letter which was either uploaded or sent using the API.
+
+    This task calls the `recreate_pdf_for_precompiled_letter` template preview task which recreates the
+    PDF for a letter which is already sanitised and in the letters-pdf bucket. The new file that is generated
+    will then overwrite the existing letter in the letters-pdf bucket.
+    """
+    notification = get_notification_by_id(notification_id)
```

Comment:
should we have a check here to make sure it's a precompiled letter, not a templated one?

Optional/maybe for separate PR/story: it would be cool to have a task / function that would recognise which letters are precompiled and which are templated and deal with them appropriately, what do yo think about this?

Reply:
I don't think we need to check that the letter is precompiled and not templated. The other letter tasks, for example `sanitise_letter` and `get_pdf_for_templated_letter`,  don't check that the letter type is right - we trust that the code is doing the right thing or that the person on support has used the right letter type. If someone did accidentally use this task with a templated letter, they would see an error in the logs when there's no file in the backup S3 folder but nothing would actually happen, so it wouldn't cause an issue.

I did think about having a task that recognises which letters are precompiled. One reason I didn't do it (apart from the fact that it involved writing more code!) is that the queries I have to find the different types use different end times. So my thinking was that we couldn't run a task for all letters at once, since there's not a single database query to find all affected letters.